### PR TITLE
Remove auto-load and specify data fields

### PR DESCRIPTION
**Pull request by mmaelicke**
This is my preliminary draft for data input into the tool container. Basically everything is optional, as there might well be tools that can handle a broad variety of inputs on the one end, and tool that are so specific, that there is no other way than supplying a detailed description on the other hand.

### CODE REVIEW

File: docs/input.md [hunk 1]
```diff
@@ -147,69 +147,49 @@ Note, that default parameters are only parsed if they are not set as `optional=t
 ## Data: File specification
 
 All input `Data` is described in a data block in the `/src/tool.yml` file.
-All sets of input data are collected as the **optional** `tools.<tool_name>.data` block:
+All sets of input data are collected as the **optional** `tools.<tool_name>.data` block.
+The simples declaration of input data is to list all available data files in a
```
Comment (AlexDo1) [inline docs/input.md:151]: *`simplest` or `most simple`

File: docs/input.md [hunk 2]
```diff
@@ -219,6 +199,37 @@ description: |
...
 ```
 
+#### `example`
+
+The `example` field is optional and can be used to reference a sample dataset
+for the given input, **within** the container. Data examples are a prime source 
+for your users to understand how inputs should look like and be formatted.
+
+```yaml
+example: /samples/input_name.csv
```
Comment (AlexDo1) [inline docs/input.md:209]: The idea of an `example` field is great!  
We also always wanted to include exemplary data and parameters in the /in folder of a container but I think it is also a good idea to have be more clear about that and have that extra /samples folder.

File: docs/input.md [hunk 3]
```diff
@@ -219,6 +199,37 @@ description: |
...
+  accuracy: | 
+    Describes if the tool has expectations of minimum required accurary.
+    This can involve measurement accuracy, but also expected scaling.
+  validity: | 
+    Describes which format requirements the tool has, to recognize the passed in
+    files as valid data inputs. 
+```
+
+There are additional dimensions to data quality, consistency, timeliness and 
+uniqueness. These dimensions do not apply here as general catgories.
```
Comment (AlexDo1) [inline docs/input.md:231]: *categories

File: docs/input.md [hunk 4]
```diff
@@ -219,6 +199,37 @@ description: |
...
+
+The `example` field is optional and can be used to reference a sample dataset
+for the given input, **within** the container. Data examples are a prime source 
+for your users to understand how inputs should look like and be formatted.
+
+```yaml
+example: /samples/input_name.csv
+```
+
+#### `quality`
```
Comment (AlexDo1) [inline docs/input.md:212]: The `quality` field is also a good idea, but I'm not sure what we would do with that in practice, is it just a free text or do we want to include checks of appropriate data quality or something like that when running a tool?

**Review by AlexDo1 — CHANGES_REQUESTED** [review]
Just a few typos and comments, looks good.

File: docs/input.md (hunk 4 above)
Comment (mmaelicke) [reply docs/input.md:212]: I have really no idea how we could quality check arbitrary data of any kind, any variable and any format in a meaningful way, whatsoever.
The idea is more aimed at giving the developer of a tool the possibility to highlight data quality considerations. The user can be made aware of them and client applications could implement them as warnings, or hints.

File: docs/input.md [hunk 5]
```diff
@@ -206,30 +206,15 @@ for the given input, **within** the container. Data examples are a prime source
...
-    This can involve measurement accuracy, but also expected scaling.
-  validity: | 
-    Describes which format requirements the tool has, to recognize the passed in
-    files as valid data inputs. 
-```
-
-There are additional dimensions to data quality, consistency, timeliness and 
-uniqueness. These dimensions do not apply here as general catgories.
-
+It is considered good practice to add example data and example parameterizaitons
```
Comment (AlexDo1) [inline docs/input.md:212]: *parametrization

File: docs/input.md [hunk 6]
```diff
@@ -206,30 +206,15 @@ for the given input, **within** the container. Data examples are a prime source
...
+to the `/in/` folder. At inspection time, when a client application reads the 
+`tool.yml`, this client can also access the examples in the `/in/` folder.
+At runtime, as the client application mounts data and parameterizations into the
```
Comment (AlexDo1) [inline docs/input.md:215]: *parametrization

**Review by AlexDo1 — APPROVED** [review]
Looks good, you can fix the two typos and merge.

File: docs/input.md (hunk 5 above)
Comment (mmaelicke) [reply docs/input.md:212]: I always use this spelling. You can use 'parameterization'  and 'parametrization'

**Pull request merged**